types/wlr_layer_shell_v1.c

```diff
@@ -339,10 +339,6 @@ static void layer_surface_role_commit(struct wlr_surface *wlr_surface) {
 		surface->mapped = true;
 		wlr_signal_emit_safe(&surface->events.map, surface);
 	}
-	if (surface->configured && !wlr_surface_has_buffer(surface->surface) &&
```

Comment:
This is not done in the precommit handler for layer-shell.

Reply:
Hm. It should be, though.

Reply:
Added role precommit handler for `layer-surface`.